apps/tools/components/send_invitation/index.coffee

```diff
@@ -24,32 +54,10 @@ module.exports = ($el) ->
       type: 'POST'
       data: serializer.data()
 
-    .then ->
-      $form.trigger 'reset'
-
-      $submit
-        .prop 'disabled', false
-        .text 'Sent!'
-
-      setTimeout (-> $submit.text label), 2500
-
-      track.submit 'Invitation sent from user'
+    .then onSuccess
 
     .catch ({ responseJSON: { message, description }}) ->
-      $errors
-        .show()
-        .html """
-          #{message}<br>
-          #{description}
-        """
-
-      $submit
-        .prop 'disabled', false
-        .text 'Error'
-
-      setTimeout ->
-        $submit.text label
-        $errors.empty()
-      , 5000
-
-      track.error 'Invitation not sent, try again.'
+      onFailure("#{message}<br/>#{description}")
+
+    .catch ->
+      onFailure("Something went wrong, please contact <a href='mailto:info@are.na'>info@are.na</a> if the problem persists.")
```

Comment:
Ah yeah, the catch-all is a good call. We've definitely encountered that in the wild.